src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as propTypes from 'prop-types';
+import omit = require('lodash/omit');
+
+import FormFieldSpinnerUp from 'wix-ui-icons-common/system/FormFieldSpinnerUp';
+import FormFieldSpinnerDown from 'wix-ui-icons-common/system/FormFieldSpinnerDown';
+
+import {
+  TimePicker as CoreTimePicker,
+  TimePickerProps as CoreTimePickerProps,
+} from 'wix-ui-core/TimePicker';
+import {AmPmOptions} from 'wix-ui-core/dist/src/components/TimePicker/constants';
+import style from './TimePicker.st.css';
+import {Size} from './constants';
+
+export interface TimePickerPropsExtended {
+  size?: Size,
+  value?: String;
+  disableAmPm?: boolean;
+  width?: string;
+}
+
+/*
+  'CoreTimePickerProps.useAmPm' prop must not be included in TimePickerProps.
+  Use type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>> to omit it from CoreTimePickerProps.
+  TODO: after typescript is updated to ^2.9 omit 'useAmPm' prop
+ */
+export type TimePickerProps = CoreTimePickerProps & TimePickerPropsExtended;
+
+const defaultProps = {
+  size: Size.medium,
+  disableAmPm: false,
+  error: false,
+  tickerUpIcon: <FormFieldSpinnerUp/>,
+  tickerDownIcon: <FormFieldSpinnerDown/>,
+};
+
+function getInputWidthState(timePickerWidth: string | undefined, size: Size, disableAmPm: boolean): string {
+  if (timePickerWidth) {
+    return '';
+  }
+
+  const states: string[] = [size];
+  if (!disableAmPm) {
+    states.push('ampm');
+  }
+
+  return states.join('_');
+}
+
+
+
+export class TimePicker extends React.PureComponent<TimePickerProps> {
+  static displayName = 'TimePicker';
+  static defaultProps = defaultProps;
+
+  static propTypes = {
+    ...omit(CoreTimePicker.propTypes, 'useAmPm'),
+    size: propTypes.oneOf(Object.keys(Size)),
+    value: propTypes.string,
+    disableAmPm: propTypes.bool,
+    width: propTypes.string
+  };
+
+  render() {
+    const {value, disableAmPm, size, error, width} = this.props;
+    const coreTimePickerProps = {
+      ...omit(this.props, 'size', 'value', 'disableAmPm'),
```

Comment:
omit `width` as well?

Reply:
Why does this component have a `width` prop at all?
And why does it overwrite the `style` prop altogether?

Reply:
i explained `width` in another comment

> And why does it overwrite the style prop altogether?

by this i guess you mean this line

https://github.com/wix/wix-ui-backoffice/blob/ccb8a5942b4ccb3bf4bbe50325821695267cf676/src/components/TimePicker/TimePicker.tsx#L77

wix-ui-core component is the one that's open for customization, whereas wix-ui-backoffice is not meant for that, it must be strict enough to fit backoffice applications and not more.

if we see there's a need for custom styling for some reason, we can add
it later but now i don't see reason why it's needed.

we want to avoid developers abusing components and/or coming up with
their own design. wix-ui-backoffice follows design guidelines made by UX
guild.

Reply:
It's going to be used outside of pure backoffice applications as well - the settings panel and the content manager. Both are very close to backoffice styles but at least the CM still needs some customizations (haven't started working on the panel yet so can't say for sure).

What I can overwrite with st.css I do, but there is one thing that is only available js: the width and height which change according to the grid view. Therefore, I must be able to pass them on through the `style` prop.

Reply:
wix-ui-backoffice, at least according to current plan, is the thing that should implement sketch design from UX. designers are very sensitive about what props we expose and how much customization we allow, because developers tend to overuse them and eventually break design rules.

so, wix-ui-backoffice should be pretty strict and not allow open things like `style`.
it makes perfect sense to allow `style` in wix-ui-core however, cause that one is core and not theme.

unless @shlomitc says otherwise, i'm not going to change it.

Reply:
Their entire team is out on a fun day so I guess it'll wait for Sunday. Have a nice weekend Arijus 😄

Reply:
we have a funday tomorrow too! :)